client/src/components/ViewPostDialog.tsx

```diff
@@ -71,15 +72,17 @@ function MoreOptions(props: {
   };
 
   const reportPost = () => {
-    // TODO need to fix spam / backend interactions
-
     api
       .reportPost(props.postID)
       .then((res) => {
+        props.didUserReport = true;
         if (res.status === 204) {
           setMsg('Post has been reported.');
+        } else if (res.status === 200) {
+          setMsg('Post has been reported and deleted.');
```

Comment:
After you fix the merge conflicts with routing changes, can you redirect to dashboard? Or navigate(-1)? Since the page will no longer exist